src/rt.rs

```diff
@@ -89,6 +90,9 @@ pub fn init() -> Result<(), glib::BoolError> {
         return Ok(());
     } else if is_initialized() {
         panic!("Attempted to initialize GTK from two different threads.");
+    } else if cfg!(target_os = "macos") && thread::current().name() != Some("main") {
```

Comment:
There's nothing preventing you from starting a new thread with the name "main" :)

Ideally you'd use the `pthread_main_np()` API here, that's more reliable. You don't have to depend on some external crate for that, just do

```rust
extern "C" {
    unsafe fn pthread_main_np() -> i32;
}
```

And then call that in here

Reply:
Sounds good! Thanks